.editorconfig

```diff
@@ -0,0 +1,21 @@
+; 4 space indentation (no size specified)
+[*]
+charset = utf-8
+end_of_line = lf
+indent_style = tab
+indent_size = 4
+[*.{php,c,h,m4,w32,md}]
+trim_trailing_whitespace = true
+[*.{phpt,php}]
+indent_style = space
+[*.xml]
+indent_size = 1
+indent_style = space
+[*.m4]
+indent_size = 2
+indent_style = space
+[*.w32]
+charset = utf-8
+end_of_line = crlf
```

Comment:
Is this really needed and not handled by git checkout?

Reply:
> Is this really needed and not handled by git checkout?

I'm not sure what you mean.

`*` was added to use line feed for all other types of files (e.g. brand new files added by contributors), this is added to override that for the line endings currently used by config.w32. (`\r\n`)

Some editors/IDEs do a bad job with pasting in snippets copied from elsewhere and can end up with mixes of newline types

Reply:
Well, why is this commit changing that file? That shouldn't happen.

Reply:
Most likely end_of_line just shouldn't be specified in editorconfig at all, and left up to git.

Reply:
https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings#per-repository-settings are you referring to this feature of git? https://github.com/krakjoe/apcu doesn't have a `.gitattributes` file.

Reply:
I'm referring to the general core.autocrlf setting, which people can use to convert LF to CRLF in Windows checkouts, if necessary.

Reply:
Removed the end_of_line override